spec/components/radio_spec.rb

```diff
@@ -148,6 +148,37 @@ def component_name
     assert_select "legend h1", "What is your favourite skittle?"
   end
 
+  it "renders radio-group with a legend" do
+    render_component(
+      name: "radio-group-legend",
+      heading: "What is your favourite smartie?",
+      heading_level: 0,
+      items: [
+        { label: "Red", value: "red" },
+        { label: "Blue", value: "blue" },
+      ],
+    )
+    assert_select ".govuk-radios"
+    assert_select "legend", "What is your favourite smartie?"
+    assert_select "legend span.govuk-fieldset__heading", true
```

Comment:
This test should probably also check that `assert_select "legend h2", false`.

Reply:
Makes sense